src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/best-practices/opentelemetry-otlp.mdx

```diff
@@ -0,0 +1,283 @@
+
+- `OTEL_BSP_*` for spans
+- `OTEL_METRIC_EXPORT_*` for metrics
+- `OTEL_BLRP_*` for logs
+
+If using the collector, the [batch processor](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor) controls batch size.
+
+Additionally, you should enable compression to reduce payload size and limit the likelihood of encountering payload size limits. New Relic supports `gzip` and `zstd` compression. `zstd` compression is higher performance and recommended if your exporter supports it. See [compression comparison](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configgrpc/README.md#compression-comparison) for more details on benchmark information.
+
+The mechanism to configure the endpoint will vary, but OpenTelemetry language SDKs generally support setting the `OTEL_EXPORTER_OTLP_COMPRESSION=gzip` environment variable (see [OpenTelemetry docs](https://opentelemetry.io/docs/specs/otel/protocol/exporter/) for more info).
+
+If using the collector, `gzip` is the default compression, but `zstd` can be optionally configured.
+
+## Config: Retry [#retry]
+
+Requirement level: **Recommended**
+
+In order to send OTLP data to New Relic, you should configure your OTLP exporter to retry when transient errors occur. The internet is unreliable, and failure to retry increases the likelihood of data loss.
+
+The mechanism to configure retry will vary. Some OpenTelemetry SDKs may have language specific environment variables (for example [java supports](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure) setting `OTEL_EXPERIMENTAL_EXPORTER_OTLP_RETRY_ENABLED=true`), but there is no general mechanism. Programmatic configuration may be required.
+
+If using the collector, the `otlphttpexporter` and `otlpexporter` retry by default. See [exporterhelper](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md) for more details.
+
+## Config: Metric Aggregation Temporality
+
+Requirement level: **Recommended**
+
+In order to send OTLP metric data to New Relic, you should configure your OTLP metrics exporter to prefer delta [aggregation temporality](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#temporality). While New Relic supports cumulative aggregation temporality, the New Relic metrics architecture is generally a delta metrics system. Using the default cumulative setting will generally incur more memory usage from SDKs and result in high data ingest.
+
+The mechanism to configure the endpoint will vary, but OpenTelemetry language SDKs generally support setting the `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE=delta` environment variable (see [OpenTelemetry docs](https://opentelemetry.io/docs/specs/otel/metrics/sdk_exporters/otlp/) for more info). If manually setting temporality as a function of instrument kind, configure as follows:
```

Comment:
@jack-berg This sentence is a bit confusing: ` If manually setting temporality as a function of instrument kind, configure as follows:`

Can you clarify where they configure this? Maybe provide an example?

Reply:
I reworded. For reference, this "as a function of instrument kind" comes directly from the [spec](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#metricreader):

> The default output temporality (optional), a function of instrument kind. If not configured, the Cumulative temporality SHOULD be used.

Probably not a good idea to include an example because the details will vary for each language. Best case it that the language supports setting `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE=delta` as defined above. That's the simplest and best chance of consistency across languages.